Main.hs

```diff
@@ -11,28 +11,32 @@ import qualified Data.List as L
 import Text.Printf (printf)
 
 multByIndex :: [Int] -> [Int]
-multByIndex = undefined
+multByIndex xs = zipWith (*) xs [0..]
+
 
 powerByIndex :: [Int] -> [Int]
-powerByIndex = undefined
+powerByIndex xs = zipWith (^) xs [0..]
 
 productOfDifference :: [Int] -> Int
-productOfDifference = undefined
+productOfDifference [] = error "The list is too short"
+productOfDifference [_] = error "The list is too short"
+productOfDifference xs = foldl (*) 1 (zipWith (flip (-)) (tail xs) xs)
 
 isSorted :: [Int] -> Bool 
-isSorted xs = undefined
+isSorted [] = True
+isSorted xs = and (zipWith (>=) (tail xs) xs)
 
 countElement :: Int -> [Int] -> Int 
-countElement = undefined
+countElement target = foldl (\acc x -> acc + (if x == target then 1 else 0)) 0
```

Comment:
This will work, but a more idiomatic way to do it would involve `filter`. Can you try to rewrite this function?